aten/src/ATen/native/cuda/TensorModeKernel.cu

```diff
@@ -13,18 +14,27 @@
 #include <thrust/iterator/constant_iterator.h>
 #include <thrust/sequence.h>
 #include <thrust/sort.h>
-#include <ATen/cuda/detail/IndexUtils.cuh>
-#include <ATen/native/cuda/TensorModeKernel.cuh>
-#include <ATen/cuda/ThrustAllocator.h>
 
 namespace at {
 namespace native {
 
+static inline int64_t ensure_nonempty_dim(int64_t dim) {
```

Comment:
rather than duplicating these, should you extract them from ReduceOpsUtils.h into a new header that only depends on TensorBase? ReduceOpsUtils.h ought to be able to depend on that to reduce churn, no?